src/renderer/components/+catalog/detail-item-registrations.injectable.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable, lifecycleEnum } from "@ogre-tools/injectable";
+import { computed } from "mobx";
+import rendererExtensionsInjectable from "../../../extensions/renderer-extensions.injectable";
+import type { CatalogEntity } from "../../../common/catalog";
+import { orderBy } from "lodash/fp";
+import type { CatalogEntityDetailRegistration } from "./catalog-entity-detail-registration";
+import coreDetailItemRegistrationsInjectable from "./core-detail-item-registrations.injectable";
+
+const detailItemRegistrationsInjectable = getInjectable({
+  instantiate: (di) => {
+    const extensions = di.inject(rendererExtensionsInjectable);
+    const coreRegistrations = di.inject(coreDetailItemRegistrationsInjectable);
+
+    return computed(() => {
+      const extensionRegistrations = extensions
+        .get()
+        .flatMap((extension) => extension.catalogEntityDetailItems);
+
+      return orderByPriority([...coreRegistrations, ...extensionRegistrations]);
+    });
+  },
+
+  lifecycle: lifecycleEnum.singleton,
+});
+
+export default detailItemRegistrationsInjectable;
+
+const orderByPriority = (
+  registrations: CatalogEntityDetailRegistration<CatalogEntity>[],
+) =>
+  orderBy(
+    "priority",
+    "desc",
+    registrations.map(({ priority = 50, ...rest }) => ({ priority, ...rest })),
+  );
```

Comment:
This is something we do a lot within the extension API, can we try and factor it out into a common utility?

Reply:
Nothing to be shared here. `orderBy` would be it, but it's already reusable. Extracted to separate function just to make use-place cleaner.

Reply:
No it is shared, we have a bunch of types that extend `{ priority?: number }`, so I could imagine the following function

```ts
interface PrioritySortable {
  priority?: number;
}

export function orderByPriority<T extends PrioritySortable>(objs: T[]) {
  const withDefaultPriority = objs.map(({ priority = 50, ...rest }) => ({ priority, ...rest }));

  return orderBy("priority", "desc", withDefaultPriority);
}
```

Reply:
Sure, as long as there is no other usages which require default priority other than `50`. 

And when we start sharing functions it becomes even more important that function does only one thing and name doesn't lie. In example you provided function lies and does two things.

1. Adds default priority
2. Orders by priority

Just to keep in mind in future, since it feels acceptable in this case.

I'll do the change.

Reply:
Done. Keep in mind that base of this branch changed. Do not merge until #4817 has been merged.

Reply:
Why was the branch changed if that PR doesn't also use the function?

Reply:
This PR makes the change for both.

Reply:
That is weird, why have it so that we merge a PR and then immediately merge another PR that changes it?

Reply:
Because both of these need the same code. One has to be merged first so other can use it. I don’t think PR process should cripple refactoring. 

End result is same and both of these are relatively small PRs.

Reply:
I would have just thought that adding the function in that PR and then just using it in this PR would make more sense and have a cleaner history.